### ISSUE TITLE
Project Status: Maintenance Reboot

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [mochajs/mocha#5027]
👋 Hi all! @Uzlopak, @voxpelli, and I are happy to announce that we've picked up maintenance of the Mocha organization and packages. That primarily includes the [`mocha`](https://www.npmjs.com/package/mocha) test framework.

Shoutout and thanks to @boneskull for inviting us on and helping set up many of the documentation and processes still in use today, as well as the other recent Mocha core team members: @giltayar, @juergba, @Munter, and @outsideris.

## Context

Mocha was not actively maintained for most of 2023. It was maintained off-and-on through 2022. This is natural in open source projects: because they are run by volunteers, maintenance often is dependent on personal voluntary availability in out-of-work hours.

The three of us "reboot" maintainers are joining on to:

* Get the project back into active maintenance, including regularly triaged issues and reviewed pull requests
* Update the project's tooling to reduce the overhead of maintenance
* Update Mocha as needed to stay up-to-date with the surrounding JavaScript ecosystem

### Project Charter

Mocha has been a core part of the JavaScript ecosystem for over a decade. Per its [`PROJECT_CHARTER.md`](https://github.com/mochajs/mocha/blob/ee2b70e8e82b0b67347d41028fa45521d43633d9/PROJECT_CHARTER.md):

> **Mocha** is a unopinionated, general-purpose testing framework for the JavaScript community. **Mocha** favors flexibility over rigidity, stability over disruption, and deliberation over agility. **Mocha** aims to do one thing, and do it well. **Mocha** will strive to evolve with the needs of the community, but will make every effort not to abandon its current users in doing so.

We intend to keep to this project charter. Mocha is positioned differently from other popular test frameworks in the ecosystem. Notably:

* Higher-level frameworks [Jest](https://jestjs.io) and [Vitest](https://vitest.dev) provide more abstractions and utilities on top of testing
* [Node's built-in runner](https://nodejs.org/api/test.html) integrates closely with Node itself at a lower level

We aim to keep Mocha positioned as a stable test framework, minimizing breaking changes or significant new feature additions. Our priority is for the existing users who do not want to be broken by or have to spend time reacting to changes.

## Next Steps

Our current plan is to:

- [x] Discuss this pinned issue and announce it
- [ ] Tidy up the existing contributing and maintenance documentation and processes, such as issue and PR templates
    - [ ] Migrate https://gitter.im/mochajs/contributors to a new Mocha Discord server with a `#help` Q&A channel
    - [ ] Publicly propose and get approved an invoicing system for https://opencollective.com/mochajs, then reach out to previous maintainers to compensate them for their time
- [ ] Review and triage existing issues and PRs
- [ ] Add tooling such as automatic security updates and static analysis to help with long-term maintenance

Items might be tackled out of order. You can see the full list of work items and their statuses on the [public Maintenance Reboot project board](https://github.com/orgs/mochajs/projects/6).

### Out of Scope

Note also that we're new to maintaining this project. We'll need to ramp up over the next few months. Don't expect any major overhauls anytime soon.

We don't intend to make any major breaking changes unless absolutely necessary for the project. We will drop support for [EOL Node versions](https://github.com/nodejs/release#release-schedule) in occasional breaking changes well after those Node versions go EOL.

We don't intend to overhaul any currently-useful resources such as https://mochajs.org or https://github.com/mochajs/mocha/wiki. Though, we will happily accept issue reports if those resources need improvement.

---

We're excited to help keep Mocha going forward as a project. Please do post feedback if you have any questions or suggestions. Cheers all! ❤️

**Comment by giltayar** [comment]
Nice! I helped with the esm part, and while not a big contributor, would be glad to help in some way.

**Comment by marvinhagemeister** [comment]
Excited about these news! To this day mocha remains my favourite testing framework and I'm using it quite extensively. I always come back to it because it just works how you expect it to and it's execution speed.

**Comment by outsideris** [comment]
Awesome! I will help you if there are something I can help you. It's great to see the project revitalized thanks to you guys.

**Comment by christian-bromann** [comment]
Mocha is an essential framework integration used by many other testing frameworks as well, like WebdriverIO, and I am happy to see it continue to be maintained and remain the most popular framework in the space. If there is anything I can do to help, please reach out!

**Comment by cozzbie** [comment]
Mocha is still the best runner out there. It will be an honor to be a part of keeping this going.

**Comment by Uzlopak** [comment]
The goal is first to implement a reliable CI with all the fancy stuff. Then we can start merging PRs ;).

**Comment by JoshuaKGoldberg** [comment]
> timeline

None yet, I'm afraid. Perhaps weeks or months?

**Comment by JoshuaKGoldberg** [comment]
Thanks for the enthusiasm and ❤️s everyone! Things are going well so far. We've trimmed about a quarter of the backlog out, triaged much of the rest of it, and have started working on codebase tooling. We're getting a more familiar with the project as a result.

I updated the _Next Steps_ checklist just now with a bit more detail. Our plan is to ship a new minor release soon with no features or fixes in it - to test that we can do releases at all. After that, we'll start being more aggressive on merging PRs that actually change Mocha for the better. 🚀

**Comment by JoshuaKGoldberg** [comment]
https://www.npmjs.com/package/mocha/v/10.3.0-preminor.0: is released under the `next` tag. You can install it with `npm i mocha@next`. 🚀

We'll wait a week or so to be careful, then release it as a `10.3.0` / `latest`.

**Comment by JoshuaKGoldberg** [comment]
https://www.npmjs.com/package/mocha/v/10.3.0-preminor.0: is released under the `next` tag. You can install it with `npm i mocha@next`. 🚀

```plaintext
npm i mocha@next
```

We'll wait a week or so to be careful, then release it as a `10.3.0` / `latest`.

**Comment by JoshuaKGoldberg** [comment]
✨ https://www.npmjs.com/package/mocha/v/10.3.0 ✨

```shell
npm i mocha@latest
```

**Comment by voxpelli** [comment]
Yay @JoshuaKGoldberg 🎉 🎉 🎉

**Comment by JoshuaKGoldberg** [comment]
> Review open PRs

All open PRs should now have a review on them! They're each at one of the following statuses:

* [`status: blocked`](https://github.com/mochajs/mocha/pulls?q=is%3Apr+is%3Aopen+label%3A%22status%3A+blocked%22): waiting on something else, such as an issue discussion or us not being ramped up to a `semver-major` change yet
* [`status: in triage`](https://github.com/mochajs/mocha/pulls?q=is%3Apr+is%3Aopen+label%3A%22status%3A+in+triage%22): one of us approved (🥳!) and we're waiting on >=1 other maintainer to also review
* [`status: waiting for author`](https://github.com/mochajs/mocha/pulls?q=is%3Apr+is%3Aopen+label%3A%22status%3A+waiting+for+author%22): at least one maintainer posted a review and is waiting for the author to respond to / resolve feedback

We'll start work on releasing approved features & fixes in new Mocha versions soon. 🚀

**Comment by danon** [comment]
You should also take a look at https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/mocha because currently it's a pain to use.

For example, I use mocha with `yarn` and run tests locally, not with browser. I like the fact that I can import `test`/`suite` with EsModules, that's great, but in DefinitelyTyped there are type script declarations that suggest that global `test` and `suite` exists, which is misleading.

I would love to see proper types.

**Comment by cozzbie** [comment]
I just want to say...to everyone who has revived and is now taking part, thank you so much. I remember the day when this was posted and seeing what has happened and contributions made till date literally has made me teary. Thank you all for reviving this beautiful tool.

**Comment by mark-wiemer** [comment]
We're still maintaining this, of course! Just closing this as it's been a couple years now.

Also, hi, I'm Mark, I started contributing in January 2025 and became a maintainer in August 2025 :)
